core/modules/replica/HttpIngestModule.cc

```diff
@@ -0,0 +1,1465 @@
+    // option for 'localhost' and tries to connect via UNIX socket.
+    cssConfig["hostname"] =
+            config->qservMasterDatabaseHost() == "localhost" ?
+                "127.0.0.1" :
+                config->qservMasterDatabaseHost(),
+    cssConfig["port"] = to_string(config->qservMasterDatabasePort());
+    cssConfig["username"] = "root";
+    cssConfig["password"] = Configuration::qservMasterDatabasePassword();
+    cssConfig["database"] = "qservCssData";
+
+    auto const cssAccess =
+            css::CssAccess::createFromConfig(cssConfig, config->controllerEmptyChunksDir());
+    if (not cssAccess->containsDb(databaseInfo.name)) {
+
+        // First, try to find another database within the same family which
+        // has already been published, and the one is found then use it
+        // as a template when registering the database in CSS.
+        //
+        // Otherwise, create a new database using an extended CSS API which
+        // will allocate a new partitioning identifier.
+
+        bool const allDatabases = false;
+        bool const isPublished = true;
+        auto const databases = config->databases(databaseFamilyInfo.name, allDatabases, isPublished);
+        if (not databases.empty()) {
+            auto const templateDatabase = databases.front();
+            cssAccess->createDbLike(databaseInfo.name, templateDatabase);
+        } else {
+
+            // This parameter is not used by the current implementation of the CSS API.
+            // However, we should give it some meaningless value in case of the implementation
+            // will change. (Hopefully) this would trigger an exception.
+            int const unusedPartitioningId = -1;
+
+            css::StripingParams const stripingParams(
+                    databaseFamilyInfo.numStripes,
+                    databaseFamilyInfo.numSubStripes,
+                    unusedPartitioningId,
+                    databaseFamilyInfo.overlap
+            );
+            string const storageClass = "L2";
+            string const releaseStatus = "RELEASED";
+            cssAccess->createDb(databaseInfo.name, stripingParams, storageClass, releaseStatus);
+        }
+    }
+
+    // Register tables which still hasn't been registered in CSS
+    
+    for (auto const& table: databaseInfo.regularTables) {
+        if (not cssAccess->containsTable(databaseInfo.name, table)) {
+
+            // Neither of those groups of parameters apply to the regular tables,
+            // so we're leaving them default constructed. 
+            css::PartTableParams const partParams;
+            css::ScanTableParams const scanParams;
+
+            cssAccess->createTable(
+                databaseInfo.name,
+                table,
+                databaseInfo.schema4css(table),
+                partParams,
+                scanParams
+            );
+        }
+    }
+    for (auto const& table: databaseInfo.partitionedTables) {
```

Comment:
I would have expected a space between `table` and the colon; `...table : databaseInfo...` but 🤷‍♂

Reply:
Well, this is an area which already caught my attention on a number of occasions when I was reviewing your code. And I was always wondering if the way you were doing it was right ... or wrong. So Now I wanted to finally settle the case 😹 and did a little research on this subject. First I looked at the LSST Style Guide for C++ hoping to find an example. Nothing specific for the C++11's `range for` found in there. Then I look at the documentation for this in here:
https://en.cppreference.com/w/cpp/language/range-for
And it turns out you win here!
I'm tempted to modify this file right away. Unfortunately, things get complicated by a fact that I have hundreds of places in my code which I would have to fix. And it would be silly to modify just this file w/o touching others. So, let's leave this till to a refactoring to be performed later. At the mean time (so that I wouldn't forget about tis) I have registered this case at my *TODO* list. Please, look at the very bottom of the document:
https://confluence.lsstcorp.org/display/DM/Completed%2C+on-going+and+proposed+improvements